### PR TITLE
[#008] Provide an optional configuration to set the erlang_ls server

### LINKED ISSUE #8: Feature request: add setting for server path
At the moment the `serverPath` is hardcoded as

```typescript
let serverPath = context.asAbsolutePath(path.join('erlang_ls', '_build', 'default', 'bin', 'erlang_ls'));
```

Make this a vscode setting for the plugin, so it is easy to set up an alternative (development) `erlang_ls` server.

### PR DESCRIPTION
**Pull request by alanz**
Closes #8

### CODE REVIEW

File: client/src/client.ts [hunk 1]
```diff
@@ -25,6 +25,11 @@ export async function activate(context: ExtensionContext) {
         path.join('erlang_ls', '_build', 'default', 'bin', 'erlang_ls')
     );
 
+    let executablePath = workspace.getConfiguration('erlang_ls').executable_path;
```
Comment (adam-becker) [inline client/src/client.ts]: I had a local branch started for this but my Googling failed me when it came to finding how to access extension configurations.

Comment (alanz) [reply client/src/client.ts]: I copied and pasted from the haskell one, where the feature was added by a kind contributor.

The interesting thing is that the config json blob goes to the language server too when it changes.

Comment (robertoaloi) [reply client/src/client.ts]: Why adding an extra variable here? Wouldn't it be nicer to just have `serverPath`, to specify the `.../_build/default/bin/erlang_ls` path as the default one in the config and allow the user to overwrite that one?

Comment (adam-becker) [reply client/src/client.ts]: @robertoaloi That's a good point, I would only hesitate on the directory separator (`/` or `\`) to make sure this works for Windows. Although you could argue that the path could then be fixed by a user if it didn't, I like when things just work out of the box.

**Comment by robertoaloi** [comment]
Superseded by #22

**Pull request closed without merging**